### PR TITLE
examples: athena terraform

### PR DESCRIPTION
**Pull request by tobiaszheller**
This PR adds terrafrom script necessary to bootstrap infra for self-hosted deployments of teleport.

Part of https://github.com/gravitational/teleport.e/issues/894
RFD: https://github.com/gravitational/teleport/blob/master/rfd/0118-scalable-audit-logs.md

### CODE REVIEW

File: examples/athena/athena.tf [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+    }
+  }
+}
+
+resource "aws_sqs_queue_policy" "audit_policy" {
+  queue_url = aws_sqs_queue.audit_queue.url
+  policy    = data.aws_iam_policy_document.audit_policy.json
+}
+
+resource "aws_s3_bucket" "long_term_storage" {
```
Comment (logand22) [inline examples/athena/athena.tf:111]: The inline attributes are deprecated.

**Review by logand22 — COMMENTED** [review]
This is looking good. I wonder if we should simplify this by removing things like the kms keys and extra configuration of the s3 buckets and instead provide the minimal terraform required to get them up and running and add a note about it not being _production_ ready. 

I also recommend running this through a security linter such as tfsec or trivy to see if it returns anything (If you want a complete setup).

File: examples/athena/athena.tf [hunk 2]
```diff
@@ -0,0 +1,281 @@
+provider "aws" {
+  region = var.aws_region
+}
+
+resource "aws_kms_key" "audit_key" {
```
[the remark this answers is not in the document]
Comment (tobiaszheller) [reply examples/athena/athena.tf:7]: done

File: examples/athena/athena.tf (hunk 1 above)
Comment (tobiaszheller) [reply examples/athena/athena.tf:111]: moved it to separate entities

File: examples/athena/athena.tf [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+  key_id = aws_kms_key.audit_key.id
+  policy = jsonencode({
+    Statement = [
+      {
+        Action = [
+          "kms:*"
+        ]
+        Effect = "Allow"
+        Principal = {
+          AWS = "${element(split(":", aws_sns_topic.audit_topic.arn), 4)}"
```
[the remark this answers is not in the document]
Comment (tobiaszheller) [reply examples/athena/athena.tf]: thanks! done

**Comment by tobiaszheller** [comment]
> This is looking good. I wonder if we should simplify this by removing things like the kms keys and extra configuration of the s3 buckets and instead provide the minimal terraform required to get them up and running and add a note about it not being _production_ ready.

I think we should keep it secure by default.

> I also recommend running this through a security linter such as tfsec or trivy to see if it returns anything (If you want a complete setup).

Used `tfsec` and applied fixes, only thing which left is medium `Bucket does not have logging enabled`

**Comment by tobiaszheller** [comment]
@logand22 PTAL

**Review by logand22 — APPROVED** [review]
Assuming this works and has been tested, it LGTM. 

I would say that if you leave object lock and lifecycle configuration in I'd recommend adding comments to them as someone new to this wouldn't know that they aren't required.

File: examples/athena/athena.tf [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+
+resource "aws_s3_bucket_public_access_block" "long_term_storage" {
+  bucket                  = aws_s3_bucket.long_term_storage.id
+  block_public_acls       = true
+  block_public_policy     = true
+  ignore_public_acls      = true
+  restrict_public_buckets = true
+}
+
+resource "aws_s3_bucket_object_lock_configuration" "long_term_storage" {
```
[the remark this answers is not in the document]
Comment (tobiaszheller) [reply examples/athena/athena.tf]: removed it and added comment inside bucket that it's recommended

**Comment by tobiaszheller** [comment]
@lxea @smallinsky PTAL

**Review by smallinsky — APPROVED** [review]

**Review by tigrato — APPROVED** [review]

**Pull request merged**